tests/e2e/parallel/main_test.go

```diff
@@ -2749,6 +2753,484 @@ func TestCustomRuleValidation(t *testing.T) {
+	if err != nil {
+		t.Fatalf("Failed to get TailoredProfile: %v", err)
+	}
+
+	expectedErrorContent := "cannot mix CustomRules and regular Rules"
+	if !strings.Contains(tpWithError.Status.ErrorMessage, expectedErrorContent) {
+		t.Fatalf("Expected error message to contain '%s', but got: %s", expectedErrorContent, tpWithError.Status.ErrorMessage)
+	}
+	t.Logf("Error message correctly indicates mixed rule types: %s", tpWithError.Status.ErrorMessage)
+
+	// Step 5: Create a TailoredProfile with only CustomRules (should work)
+	tpValidName := fmt.Sprintf("%s-tp-valid", testName)
+	tpValid := &compv1alpha1.TailoredProfile{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      tpValidName,
+			Namespace: testNamespace,
+			Annotations: map[string]string{
+				compv1alpha1.DisableOutdatedReferenceValidation: "true",
+			},
+		},
+		Spec: compv1alpha1.TailoredProfileSpec{
+			Title:       "CustomRules Only Test",
+			Description: "This profile correctly uses only CustomRules",
+			EnableRules: []compv1alpha1.RuleReferenceSpec{
+				{
+					Name:      customRuleName,
+					Kind:      "CustomRule",
+					Rationale: "Ensure containers are not privileged",
+				},
+			},
+		},
+	}
+
+	err = f.Client.Create(context.TODO(), tpValid, nil)
+	if err != nil {
+		t.Fatalf("Failed to create valid TailoredProfile: %v", err)
+	}
+	defer f.Client.Delete(context.TODO(), tpValid)
+
+	// Should be ready since it only has CustomRules
+	err = f.WaitForTailoredProfileStatus(testNamespace, tpValidName, compv1alpha1.TailoredProfileStateReady)
+	if err != nil {
+		t.Fatalf("Valid TailoredProfile did not become ready: %v", err)
+	}
+	t.Logf("TailoredProfile %s with only CustomRules is ready as expected", tpValidName)
+
+	// Step 6: Create a TailoredProfile with only regular Rules (should work)
+	tpRegularName := fmt.Sprintf("%s-tp-regular", testName)
+	tpRegular := &compv1alpha1.TailoredProfile{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      tpRegularName,
+			Namespace: testNamespace,
+		},
+		Spec: compv1alpha1.TailoredProfileSpec{
+			Title:       "Regular Rules Only Test",
+			Description: "This profile correctly uses only regular Rules",
+			EnableRules: []compv1alpha1.RuleReferenceSpec{
+				{
+					Name:      "ocp4-cluster-version-operator-exists",
+					Rationale: "Make sure cluster version operator exists",
+				},
+				{
+					Name:      "ocp4-file-owner-scheduler-kubeconfig",
+					Kind:      "Rule", // Explicitly set Kind to Rule
+					Rationale: "Ensure proper ownership of scheduler kubeconfig",
+				},
+			},
+		},
+	}
+
+	err = f.Client.Create(context.TODO(), tpRegular, nil)
+	if err != nil {
+		t.Fatalf("Failed to create regular TailoredProfile: %v", err)
+	}
+	defer f.Client.Delete(context.TODO(), tpRegular)
+
+	// Should be ready since it only has regular Rules
+	err = f.WaitForTailoredProfileStatus(testNamespace, tpRegularName, compv1alpha1.TailoredProfileStateReady)
```

Comment:
Interesting - this doesn't seem to ever reach a ready state. Is it because we're not passing in a Kind on line 2920?

Reply:
I need to chose a different platform rule instead in the test